### PR TITLE
Fixes #18138: Split Number in Integer and Float

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/18138

### CODE REVIEW

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Review by gpoblon — APPROVED** [review]

**Comment by amousset** [comment]
OK, squash merging this PR

File: rudder-lang/docs/types.adoc [hunk 1]
```diff
@@ -4,20 +4,26 @@
...
 * an unescaped string is delimited by `"""`
 * interpolation has the following syntax: `${...}`
 * supported escape sequences: `\\`, `\n`, `\r`, `\t`
 
-[#num-type]
-== num
+[#integer-type]
+== integer
+
+Internally represented by a 64-bit signed integer.
```
Comment (peckpeck) [inline rudder-lang/docs/types.adoc:16]: we will need a lot of cfengine to keep that promise

**Pull request merged**